### PR TITLE
Remember the last host used and re-use it if <60sec

### PR DESCRIPTION
**Pull request by redox**

### CODE REVIEW

**Comment by redox** [comment]
@alexandremeunier @Jerskouille would love your review 👍

File: spec/client_spec.rb [hunk 1]
```diff
@@ -997,6 +997,15 @@ def test_browse(expected, *args)
         @client.list_indexes # fallback on the second host after 5 sec (connection timeout)
         expect(start_time.to_i + 5).to be <= Time.now.to_i + 1
       end
+
+      it "should re-use the working the 2nd host after the first one failed" do
```
Comment (rayrutjes) [inline spec/client_spec.rb]: Looks like you can remove 'the' before 2nd ;)

File: lib/algolia/client.rb [hunk 2]
```diff
@@ -374,16 +377,27 @@ def delete(uri, type = :write)
 
     # This method returns a thread-local array of sessions
     def thread_local_hosts(read)
```
Comment (Jerskouille) [inline lib/algolia/client.rb]: I had a hard time following what would happen on the first call where current_index would be `nil`.
```
Thread.current[read ? :algolia_search_host_index : :algolia_host_index] ||= 0
```
would have made everything clearer here, and we could then remove the `.to_i` line 393.

**Comment by Jerskouille** [comment]
LGTM 👍

File: lib/algolia/client.rb (hunk 2 above)
Comment (redox) [inline lib/algolia/client.rb]: Arg yes indeed, using `nil.to_i` is something I've used to do.

**Pull request merged**